### PR TITLE
Issue 7 - Create builders list page

### PR DESCRIPTION
**Pull request by danitome24**
## Description

Created a page to display a list of builders and mentors. The page features:

- A section for mentors with their photos, names, and links to their personal pages.

- A section for builders with similar details.

- A dynamic counter that shows how many builders have checked in out of the total.

You can see a first template:

image

## Additional Information

- [X] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [X] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

_Closes #7_

Your ENS/address: 0xaa4C60b784E2b3E485035399bF1b1aBDeD66A60f

### CODE REVIEW

**Comment by danitome24** [comment]
TODO:

- [ ] Split into smaller components.
- [ ] Handle loading events message (display some message that is loading data).
- [ ] What we do with randomImages?

File: packages/nextjs/app/builders/page.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+const getRandomUserImage = () => {
+  const randomInt = Math.floor(Math.random() * 50);
+  const randomGender = getRandomGender();
+
+  return `https://randomuser.me/api/portraits/${randomGender}/${randomInt}.jpg`;
+};
+
+const getBuilderFromEvent = (event: any): Builder => ({
+  image: getRandomUserImage(),
```
Comment (derrekcoleman) [inline packages/nextjs/app/builders/page.tsx:21]: I wonder what else we can do for profile pictures 🤔 

Your design does a great job emphasizing the rounded portraits, so it seems silly to have random placeholders in the final version.
- Is it possible to map builders to their GitHub profiles and fetch those avatars, just like you do for mentors? 
- If not, there are also BlockieAvatars ([SE-2 docs](https://docs.scaffoldeth.io/components/BlockieAvatar)) you can modify.
- [Here's](https://github.com/BuidlGuidl/buidlguidl.com/blob/1cbcaa5a09d7bf276a550af53663cdfbab73f9d0/packages/nextjs/components/scaffold-eth/BlockieAvatar.tsx#L5) where they are used in the BuidlGuidl app!
Screenshot 2024-09-19 at 2 52 20 PM

File: packages/nextjs/app/builders/page.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    link: "https://github.com/derrekcoleman",
+    checkedIn: false,
+  },
+  {
+    name: "Philip",
+    image: "https://avatars.githubusercontent.com/u/90064316?v=4",
+    link: "https://github.com/phipsae",
+    checkedIn: false,
+  },
+];
```
Comment (derrekcoleman) [inline packages/nextjs/app/builders/page.tsx:45]: I really appreciate your inclusion of the mentors :)))

File: packages/nextjs/types/builders.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export type Builder = {
+  image: string;
+  link: string;
+  checkedIn: boolean;
+};
+
+export type Mentor = Builder & {
+  name: string;
+};
```
Comment (derrekcoleman) [inline packages/nextjs/types/builders.ts:9]: Very clean

**Review by derrekcoleman — CHANGES_REQUESTED** [review]
Great job! Very clean architecture. I left some suggestions about the pictures - lmk what you think!

File: packages/nextjs/app/builders/page.tsx (hunk 1 above)
Comment (danitome24) [reply packages/nextjs/app/builders/page.tsx:21]: I have tried to obtain the GitHubs of the builders automatically, but I haven't found a way. For this, I suppose we would need to manually keep a record of all users and their GitHubs. I have therefore decided to use BlockieAvatars for the image.

Maybe somebody can help with this and find a way to get github profiles automatically.

Comment (ishtails) [reply packages/nextjs/app/builders/page.tsx:21]: we could use ENS avatars. let me try something out...

**Comment by ishtails** [comment]
@danitome24 really nice work! i have some small changes that i'd like to contribute to this PR. could you allow me to push my commit in this branch of your fork? here's the screenshot of what i'm thinking:
- ens avatar is displayed whenever it exists
- user address is displayed below the avatar for identification
- filtered duplicate builder entries before rendering

screenshot:
Screenshot (433)

**Comment by derrekcoleman** [comment]
@ishtails you've got the right idea. Check out [this discussion](https://www.reddit.com/r/git/comments/bc4ie8/creating_a_pull_request_off_of_a_branch_of/) about how to manage multiple “remotes” for your local version of the repo. The goal is to make a PR to Dani’s fork of the repo (rather than to the main repo we’re having this discussion in).

**Comment by ishtails** [comment]
> @ishtails you've got the right idea. Check out [this discussion](https://www.reddit.com/r/git/comments/bc4ie8/creating_a_pull_request_off_of_a_branch_of/) about how to manage multiple “remotes” for your local version of the repo. The goal is to make a PR to Dani’s fork of the repo (rather than to the main repo we’re having this discussion in).

Tried forking his fork, for some reason i am not able to do that.. so i am thinking to just open a new PR for making the changes i proposed once this is merged (anyway i wanted to redesign the UI like the new home page layout). please proceed with the merge as usual. i'll pick it up once this PR is closed.

**Comment by derrekcoleman** [comment]
> Tried forking his fork, for some reason i am not able to do that.. so i am thinking to just open a new PR for making the changes i proposed once this is merged (anyway i wanted to redesign the UI like the new home page layout). please proceed with the merge as usual. i'll pick it up once this PR is closed.

Not so fast @ishtails - you can do this. Your goal is to make a PR to [his repo](https://github.com/danitome24/batch9.buidlguidl.com/tree/7-list-members-of-batch), specifically to his `7-list-members-of-batch` branch. You don't need to fork his repo if you already have this repo locally. Instead, you need to change your `remote` to his repo's URL, then pull/push there.

Resources to learn about changing your `remote`, which controls where your `git push` goes.
- [GitHub docs](https://docs.github.com/en/get-started/getting-started-with-git/managing-remote-repositories)
- [StackOverflow 1](https://stackoverflow.com/questions/2432764/how-do-i-change-the-uri-url-for-a-remote-git-repository)
- [StackOverflow 2](https://stackoverflow.com/questions/63515728/git-how-to-push-to-someones-fork)

Definitely worth learning because (a) you will encounter this again in the future and (b) even though the learning make take a moment, the actual fix is only a couple `git` commands.

**Comment by ishtails** [comment]
Please

**Comment by ishtails** [comment]
> > Tried forking his fork, for some reason i am not able to do that.. so i am thinking to just open a new PR for making the changes i proposed once this is merged (anyway i wanted to redesign the UI like the new home page layout). please proceed with the merge as usual. i'll pick it up once this PR is closed.
> 
> Not so fast @ishtails - you can do this. Your goal is to make a PR to [his repo](https://github.com/danitome24/batch9.buidlguidl.com/tree/7-list-members-of-batch), specifically to his `7-list-members-of-batch` branch. You don't need to fork his repo if you already have this repo locally. Instead, you need to change your `remote` to his repo's URL, then pull/push there.
> 
> Resources to learn about changing your `remote`, which controls where your `git push` goes.
> 
> * [GitHub docs](https://docs.github.com/en/get-started/getting-started-with-git/managing-remote-repositories)
> * [StackOverflow 1](https://stackoverflow.com/questions/2432764/how-do-i-change-the-uri-url-for-a-remote-git-repository)
> * [StackOverflow 2](https://stackoverflow.com/questions/63515728/git-how-to-push-to-someones-fork)
> 
> Definitely worth learning because (a) you will encounter this again in the future and (b) even though the learning make take a moment, the actual fix is only a couple `git` commands.

haha you were right, i was just running away xD.. got it working with your and @danitome24's help. thankyou for teaching me something new :))

**Comment by derrekcoleman** [comment]
I’m proud of your persistence! Thanks for the help, @danitome24 🤝

**Pull request merged**